node/src/components/block_proposer.rs

```diff
@@ -332,7 +336,26 @@ impl BlockProposerReady {
         if self.sets.finalized_deploys.contains_key(&hash) {
             info!(%hash, "deploy rejected from the buffer");
         } else {
-            self.sets.pending.insert(hash, deploy_or_transfer);
+            let gas_price = deploy_or_transfer.header().gas_price();
+            let payment_amount_gas = if deploy_or_transfer.is_transfer() {
+                Gas::from(self.wasmless_transfer_cost)
+            } else {
+                match Gas::from_motes(deploy_or_transfer.payment_amount_motes(), gas_price) {
+                    Some(value) => value,
```

Comment:
@afck, everything seems right here, however?

- we get the gas price from the header
- IF a transfer, we convert u64 wasmless_transfer_cost we got from the chainspec to Gas (note that we don't really need to do that for the sort itself @dwerner)
- ELSE (IF regular deploy) we convert payment amount specified in the deploy to gas amount using the per unit price

So, now I'm really confused what the problem is supposed to be

Reply:
Ah, you're right, the sorting should actually be fine.
It's still kind of wrong that there is a constant `self.wasmless_transfer_cost` in motes, but it doesn't matter here.

Reply:
@afck, it is not in motes. It's read from chainspec (which intends it to be gas, as stated there) as a u64, then, _for transfers_ converted to gas using `from`

`impl From<u32> for Gas {
    fn from(gas: u32) -> Self {
        let gas_u512: U512 = gas.into();
        Gas::new(gas_u512)
    }
}`

Reply:
Note that the regular deploy conversion to Gas used `from_motes`, rather than `from`

Reply:
Ah, right! Sorry for the confusion!